synapseclient/models/link.py

```diff
@@ -0,0 +1,423 @@
+"""Link dataclass model for Synapse entities."""
+
+import dataclasses
+from copy import deepcopy
+from dataclasses import dataclass, field, replace
+from datetime import date, datetime
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Union
+
+from synapseclient import Synapse
+from synapseclient.api import get_from_entity_factory
+from synapseclient.core.async_utils import async_to_sync, otel_trace_method
+from synapseclient.core.constants.concrete_types import LINK_ENTITY
+from synapseclient.core.utils import delete_none_keys, merge_dataclass_entities
+from synapseclient.models import Activity, Annotations
+from synapseclient.models.services.search import get_id
+from synapseclient.models.services.storable_entity import store_entity
+from synapseclient.models.services.storable_entity_components import (
+    store_entity_components,
+)
+
+if TYPE_CHECKING:
+    from synapseclient.models import (
+        Dataset,
+        DatasetCollection,
+        EntityView,
+        File,
+        Folder,
+        MaterializedView,
+        Project,
+        SubmissionView,
+        Table,
+        VirtualTable,
+    )
+
+
+@dataclass()
+@async_to_sync
+class Link:
```

Comment:
This is an implementation of the Link entity. This is used to create a pointer to another Synapse entity. In the Web UI it looks like:

<img width="1815" height="556" alt="image" src="https://github.com/user-attachments/assets/e09fa352-5494-4374-bf4b-9d49f090ea1b" />